src/webview/ui/spectrogramComponent.ts

```diff
@@ -57,20 +57,61 @@ export default class WaveFormComponent {
         const scale = (maxFreq - minFreq) / height;
 
         axisContext.font = `20px Arial`;
-        for (let i = 0; i < 10; i++) {
-            axisContext.fillStyle = "rgb(245,130,32)";
-            const x = Math.round(i * width / 10);
-            const t = i * (settings.maxTime - settings.minTime) / 10 + settings.minTime;
-            if (i !== 0) axisContext.fillText(`${(t).toFixed(2)}`, x, 18);
 
-            const freq = minFreq + i * (maxFreq - minFreq) / 10;
+        // draw horizontal axis
+        if (settings.roundTimeAxis) {
+            const [nice_t, digit]: [number, number] = AnalyzeService.roundToNearestNiceNumber((settings.maxTime - settings.minTime) / 10);
+            const x_by_t = width / (settings.maxTime - settings.minTime);
+            let t = settings.minTime;
+            let loop_cnt = 0;   // safe guard
+            do {
+                t = Math.round(t / nice_t) * nice_t;
+                let x = (t - settings.minTime) * x_by_t;
+
+                axisContext.fillStyle = "rgb(245,130,32)";
+                if (width * (5 / 100) < x  && x < width * (95 / 100)) axisContext.fillText(`${(t).toFixed(digit)}`, x, 18);     // don't draw near the edge
+
+                axisContext.fillStyle = "rgb(180,120,20)";
+                for (let j = 0; j < height; j++) axisContext.fillRect(x, j, 1, 1);
+
+                t += nice_t
+            } while (t < settings.maxTime && loop_cnt++ < 100);
```

Comment:
This part can be written as const as shown below, so there is no need to track the state of t when reading. There is also no need to worry about infinite loops.
Could you change it like this?

Also, since you will be writing the same implementation three times, I think it would be a good idea to make drawTimeAxis a method. 
There is NO need to share it with waveformComponent.
(However, I would like you to make the same modifications to the processing on that side.)

```ts
const [niceT, digit] = AnalyzeService.roundToNearestNiceNumber((settings.maxTime - settings.minTime) / 10);
const dx = width / (settings.maxTime - settings.minTime);
const t0 = Math.ceil(settings.minTime / niceT) * niceT;
const numAxis = Math.floor((settings.maxTime - settings.minTime) / niceT);
for (let i = 0; i <= numAxis; i++) {
   const t = t0 + niceT * i;
   const x = (t - settings.minTime) * dx;

   axisContext.fillStyle = "rgb(245,130,32)";
   if (width * (5 / 100) < x  && x < width * (95 / 100)) axisContext.fillText(`${(t).toFixed(digit)}`, x, 18);     // don't draw near the edge

   axisContext.fillStyle = "rgb(180,120,20)";
   for (let j = 0; j < height; j++) axisContext.fillRect(x, j, 1, 1);
}
```